lib/perl/Genome/Site/TGI/CleTest.pm

```diff
@@ -0,0 +1,203 @@
+    my $cmd1 = Genome::Config::AnalysisProject::Command::AddMenuItem->create(
+        analysis_menu_items => $menu_item1,
+        tags => $discovery_tag,
+        analysis_project => $analysis_project,
+    );
+    $cmd1->execute;
+    my $cmd2 = Genome::Config::AnalysisProject::Command::AddMenuItem->create(
+        analysis_menu_items => $menu_item1,
+        tags => $followup_tag,
+        analysis_project => $analysis_project,
+    );
+    $cmd2->execute;
+    my $cmd3 = Genome::Config::AnalysisProject::Command::AddMenuItem->create(
+        analysis_menu_items => $menu_item2,
+        tags => $germline_tag,
+        analysis_project => $analysis_project,
+    );
+    $cmd3->execute;
+    my $subject_mapping_file = Genome::Sys->create_temp_file_path;
+    my $subject_mapping = <<'SUBJECT_MAPPINGS';
+H_KA-174556-1309237	H_KA-174556-1309246				followup
+H_KA-174556-1309245	H_KA-174556-1309246				discovery
+H_KA-174556-1309246					germline
+SUBJECT_MAPPINGS
+
+    Genome::Sys->write_file($subject_mapping_file, $subject_mapping);
+    my $cmd4 = Genome::Config::AnalysisProject::SubjectMapping::Command::Import::SomaticValidation->create(
+        analysis_project => $analysis_project,
+        file_path => $subject_mapping_file,
+    );
+    $cmd4->execute;
+
+    my @instrument_data = qw(2893814999 2893815000 2893815001 2893815002 2893815003 2893815016 2893815018 2893815020 2893815023 2893815024 2893815447 2893815448 2893815449 2893815451 2893815484 2893815485 2893815489 2893815492);
+    for my $id (@instrument_data) {
+        my $instrument_data = Genome::InstrumentData->get($id);
+        Genome::Config::AnalysisProject::InstrumentDataBridge->create(
+            analysis_project => $analysis_project,
+            instrument_data => $instrument_data,
+            status => 'new',
+        );
+    }
+
+    my $cmd5 = Genome::Config::AnalysisProject::Command::Release->create(
+        analysis_projects => [$analysis_project],
+    );
+    $cmd5->execute;
+
+    my @instrument_data_objects = Genome::InstrumentData->get(id => \@instrument_data);
+    my @pre_cqid_models = Genome::Model->get();
+    my $pre_cqid_model_count = scalar @pre_cqid_models;
+    my $cmd6 = Genome::Config::Command::ConfigureQueuedInstrumentData->create(
+        instrument_data => \@instrument_data_objects,
+    );
+    $cmd6->execute;
+    my @post_cqid_models = Genome::Model->get();
+    my $post_cqid_model_count = scalar @post_cqid_models;
+    my $expected_number_of_models = 7;
+    my $actual_number_of_models = $post_cqid_model_count - $pre_cqid_model_count;
+    unless ($actual_number_of_models == $expected_number_of_models) {
+        die $self->error_message("We expected CQID to create %s models and it actually created %s\n",
+                $expected_number_of_models,
+                $actual_number_of_models);
+    }
+
+    my @models = Genome::Model->get(analysis_projects => [$analysis_project]);
```

Comment:
The `analysis_projects` property is deprecated in favour of the singular `analysis_project`.